libs/core/langchain_core/tools/base.py

```diff
@@ -605,7 +626,12 @@ async def _arun(self, *args: Any, **kwargs: Any) -> Any:
     def _to_args_and_kwargs(
         self, tool_input: Union[str, dict], tool_call_id: Optional[str]
     ) -> tuple[tuple, dict]:
-        if self.args_schema is not None and not get_fields(self.args_schema):
+        if (
+            self.args_schema is not None
+            and not isinstance(self.args_schema, dict)
```

Comment:
`not isinstance(self.args_schema, dict)` isn't logically correct. The conditional should be in terms of what `args_schema` **is** rather than what it *is not*.

e.g., 

`args_schema` is truthy, and it can be subclassed, and it's a subclass of one of basemodel or basemodelv1..

Reply:
yup, this was a hacky solution to deal w/ issubclass checks but there is a better way, replaced it